zeus/api/resources/github_repositories.py

```diff
@@ -78,7 +78,42 @@ def delete(self):
         if not repo_name:
             return self.error("missing repo_name parameter")
 
-        raise NotImplementedError
+        owner_name, repo_name = repo_name.split("/", 1)
+
+        user = auth.get_current_user()
+        repo = (
+            Repository.query.unrestricted_unsafe()
+            .filter(
+                Repository.owner_name == owner_name,
+                Repository.name == repo_name,
+                Repository.backend == RepositoryBackend.git,
+                Repository.provider == RepositoryProvider.github,
+            )
+            .first()
+        )
+
+        if not repo:
+            return self.error("repository not found")
+
+        repo_access = RepositoryAccess.query.filter_by(
+            repository_id=repo.id, user_id=user.id
+        ).first()
+
+        if repo_access.permission != Permission.admin:
+            return self.respond(
+                {"message": "Insufficient permissions to deactivate repository"}, 403
+            )
+
+        lock_key = self._get_lock_key(owner_name, repo_name)
+        with redis.lock(lock_key):
+            ItemOption.query.filter_by(
+                item_id=repo.id, name="auth.private-key"
+            ).delete()
+            RepositoryAccess.query.filter_by(repository_id=repo.id).delete()
+            db.session.delete(repo)
```

Comment:
we night consider just setting it to inactive rather than making it a hard delete

otherwise a couple of things:

1. hard delete needs to go through workers since it could have to cascade into a LOT of rows
2. giant scary warning on hard delete

Reply:
I certainly see value in hard deletes, e.g. in case of private repos. Users might forget to delete those repo deploy keys from github, so at least private keys will be removed from Zeus.

1. and 2. make perfect sense, updated 👍